### PR TITLE
Add 0.8.0 OZ contracts

### PR DESCRIPTION
**Pull request by mikiw**
## Usage related changes

- support of 0.8.0 version of OZ account contract and ERC20 contract was added, users can still use 0.5.1 with `--account-class cario0`, internal chargeable account is still using 0.5.1
- cairo 1 accounts are now predeployed the default (used to be cairo 0)

## Development related changes

- mint function implementation is missing from ERC20_Mintable_OZ_0.8.0.json, OpenZeppelin is providing [wizard](https://wizard.openzeppelin.com/cairo) with mintable feature implementation that, we used.

## Checklist:

- [x] Applied formatting - `./scripts/format.sh`
- [x] No linter errors - `./scripts/clippy_check.sh`
- [x] No unused dependencies - `./scripts/check_unused_deps.sh`
- [x] Performed code self-review
- [x] Rebased to the last commit of the target branch (or merged it into my branch)
- [x] Documented the changes
- [x] Linked the issues which this PR resolves
- [x] Checked the TODO section in README.md if this PR resolves it
- [x] Updated the tests
- [x] All tests are passing - `cargo test`

### CODE REVIEW

**Comment by mikiw** [comment]
I'm still on the 0.8.1 compilation error topic, I might update this PR to be 0.8.1 not 0.8.0

**Comment by mikiw** [comment]
PR is ready for review again. I needed to fix one test after the rebase.

And about 0.8.1 - it can be handled in new PR.

File: crates/starknet-devnet-core/src/starknet/predeployed.rs [hunk 1]
```diff
@@ -4,26 +4,23 @@ use starknet_types::contract_storage_key::ContractStorageKey;
...
-        })?;
-    let erc20_fee_contract = SystemContract::new(
-        ERC20_CONTRACT_CLASS_HASH,
-        contract_address,
-        &erc20_contract_class_json_str,
-    )?;
+pub(crate) fn create_erc20_at_address(
+    contract_address: &str,
+    path: &str,
+    hash: &str,
```
Comment (FabijanC) [inline crates/starknet-devnet-core/src/starknet/predeployed.rs:18]: Why should this now be specified if in production we will always use the same cairo 1 implementation? Shouldn't the path and the hash be hardcoded?

**Review by FabijanC — CHANGES_REQUESTED** [review]

File: crates/starknet-devnet-core/src/starknet/predeployed.rs (hunk 1 above)
Comment (mikiw) [reply crates/starknet-devnet-core/src/starknet/predeployed.rs:18]: Thanks for the review. It's a remnant of my refactor where I was mixing both versions.

Fixing...

Comment (mikiw) [reply crates/starknet-devnet-core/src/starknet/predeployed.rs]: Fixed in 94bd3f331a5ae81dadf01b0cf9efe8a451c29bfd

**Review by FabijanC — APPROVED** [review]

**Pull request merged**